codice/ddf
# DDF-3872 Installation cannot successfully complete in an isolated, offline environment due to remote schema loading

**Important!** Look at the commits. The first one is just schema files. The 2nd one has meaningful code changes. 

#### What does this PR do?
Fixes an issue where, on bundle start, `catalog-ui-search` will make web requests for schema document `<import>` dependencies that were missed the first time. The remaining schemas have been added as JAR resources and the `FilterWriter` will transform `schemaLocation` URIs to point to the local copy. 

As a direct consequence of this, `catalog-ui-search` starts up significantly faster now. And no errors should occur when an internet connection is not available. 

#### Who is reviewing it? 
@dannythk 
@mackncheesiest 
@Schachte 

#### Ask 2 committers to review/merge the PR and tag them here.
@rzwiefel  
@coyotesqrl 

#### How should this be tested?
Build the DDF zip. Disconnect from all networks before deploying. Validate this first by trying to load a web page and ensure the browser-specific "No Internet Detected" page is displayed.  

Startup, install, and the subsequent restart of DDF should occur without any schema-related exceptions in the logs. Ensure the Intrigue UI can be reached. 

#### Any background context you want to provide?
These schemas are part of Filter 2.0 and serve as a final validation step against programmatic errors during search template transformations between JSON and XML. 

#### What are the relevant tickets?
[DDF-3872](https://codice.atlassian.net/browse/DDF-3872)

#### Checklist:
- [ ] Update / Add Integration Tests (N/A)

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
